CorsixTH/Lua/entities/machine.lua

```diff
@@ -156,7 +160,23 @@ function Machine:machineUsed(room)
       local call = self.world.dispatcher:callForRepair(self, true, false, true)
       self.hospital:addHandymanTask(self, "repairing", 2, self.tile_x, self.tile_y, call)
     else -- Otherwise the task is already queued. Increase the priority to above that of machines with at least 4 uses left
-      self.hospital:modifyHandymanTaskPriority(taskIndex, 2, "repairing")
+      local subTable = self.hospital:findHandymanTaskSubtable("repairing")
+      local sound = room.room_info.handyman_call_sound
+      local earthquake = self.world.next_earthquake
+      if subTable[taskIndex].priority == 1 then
+        -- If an earthquake is happening limit the amount of machine warnings
+        if earthquake.active and earthquake.warning_timer == 0 then
+          if not earthquake.machwarn_trigger then
+            self.world.ui:playAnnouncement("machwarn.wav", AnnouncementPriority.Critical)
+            earthquake.machwarn_trigger = true -- resets with each damage phase
+          end
+        else
+          -- Must be same priority
+          self.world.ui:playAnnouncement("machwarn.wav", AnnouncementPriority.Critical)
+          if sound then self.world.ui:playAnnouncement(sound, AnnouncementPriority.Critical) end
+        end
```

Comment:
This code looks duplicate to the calls dispatcher addition.

Don't clone code. If it contains a bug, or it should be modified in the future, you have practically a guarantee that not all clones will be modified in the same way, thus creating bugs, as the same thing works subtly different depending on the code path taken.

Instead, make a function for it, and call the function from every spot. Less code, no duplication, no missing of cloned code, less bugs